cts.json

```diff
@@ -2429,6 +2446,24 @@
       "document" : [{"a": "bc"}],
       "result": []
     },
+    {
+      "name": "filter, search function, unicode char class, uppercase",
+      "selector": "$[?search(@, '\\\\p{Lu}')]",
+      "document": ["ж", "Ж", "1", "жЖ", true, [], {}],
+      "result": ["Ж", "жЖ"]
+    },
+    {
+      "name": "filter, search function, unicode char class negated, uppercase",
+      "selector": "$[?search(@, '\\\\P{Lu}')]",
+      "document": ["ж", "Ж", "1", true, [], {}],
+      "result": ["ж", "1"]
+    },
+    {
+      "name": "filter, search function, unicode, surrogate pair",
+      "selector": "$[?search(@, 'a.b')]",
+      "document": ["a\uD800\uDD01bc", "abc", "1", true, [], {}],
+      "result": ["a\uD800\uDD01bc"]
+    },
```

Comment:
Such a test should be provided in so we know which implementations do not support non-BMP Unicode.

This should be added to the comparison site, then, not the test suite.

Reply:
This deviation is about implementation bugs, not about features.

(I could also imagine that ECMAScript implementations will ignore the LS/PS problem mentioned in Section 5.3 of iregexp.)

Reply:
This may just be a .Net issue.  I may have to ignore this test, then.

Ref: https://github.com/ietf-wg-jsonpath/iregexp/issues/22#issuecomment-1510520470